src/Demos/HSP.lagda

```diff
@@ -268,65 +222,53 @@ f ⟨∘⟩ g = record  { f = (_⟨$⟩_ f) ∘ (_⟨$⟩_ g)
 \end{code}
 \fi
 
-\paragraph*{Inverses of setoid functions}
-We begin by defining an inductive type that represents the \emph{image} of a function.
-%\footnote{cf.~the \ualmodule{Overture.Setoid.Inverses} module of the \agdaalgebras library.}
+\paragraph*{Inverses}
+We frequently need to deal with inverses. These are most easily defined via the
+\emph{image} of a function.
 
 \begin{code}
-
 module _ {𝑨 : Setoid α ρᵃ}{𝑩 : Setoid β ρᵇ} where
  open Setoid 𝑩 using ( _≈_ ; sym ) renaming ( Carrier to B )
 
  data Image_∋_ (f : 𝑨 ⟶ 𝑩) : B → Type (α ⊔ β ⊔ ρᵇ) where
   eq : {b : B} → ∀ a → b ≈ f ⟨$⟩ a → Image f ∋ b
-
 \end{code}
-An inhabitant of \aod{Image} \ab f \aod{∋} \ab b is a dependent pair \AgdaPair{a}{p},
-where \AgdaTyped{a}{A} and \ab p~\as :~\ab b \af{≈} \ab f~\ab a is a proof that
-\ab f maps \ab a to \ab b.  Since the proof that \ab b
-belongs to the image of \ab f is always accompanied by a witness \AgdaTyped{a}{A}, we can
-actually \emph{compute} a range-restricted right-inverse of \ab f, as follows.
-%. For convenience, we define this
-%inverse function and give it the name \af{Inv}.
 
-\begin{code}
+An inhabitant of \aod{Image} \ab f \aod{∋} \ab b is a point \ab
+where \AgdaTyped{a}{A} and a proof \ab p~\as :~\ab b \af{≈} \ab f~\ab a that
+\ab f maps \ab a to \ab b.
+Since the witness that \ab b
+belongs to the image of \ab f is always accompanied by a concrete witness \AgdaTyped{a}{A}, we can
+\emph{compute} a range-restricted right-inverse of \ab f.  For extra certainty, we also verify
+that your witness really is an inverse.
```

Comment:
"your witness"?  I prefer, "our witness"  (we're all in this together :) ...but maybe you had a reason for saying "your"

Reply:
No strong reason, 'our' is fine